File: com.ibm.mq.ssl-wizard/src/tlswizard/samples/SSLSample.java
/*******************************************************************************
 * Copyright (c) 2005,2014 IBM Corporation and other Contributors.
 *
 * All rights reserved. This program and the accompanying materials
 * are made available under the terms of the Eclipse Public License v1.0
 * which accompanies this distribution, and is available at
 * http://www.eclipse.org/legal/epl-v10.html
 *
 * Contributors:
 *   Ian Vanstone - Initial Contribution
 *******************************************************************************/

package tlswizard.samples;

import com.ibm.mq.*; //Include the WebSphere MQ classes for Java package
import com.ibm.mq.constants.MQConstants; 

public class SSLSample {

  // define the parms
  private static String conname ;
  private static String port    ;
  private static String channel ;
  private static String qmgr    ;
  private static String sslciph ;
  private static String sslkeyr ;
  private static String sslpass ;
  
  public static void main(String args[]) {
    /****************************************************************/
    /* Check for correct number of arguments                        */
    /****************************************************************/
    if (args.length == 7) {
      conname = args[0];
      port    = args[1];
      channel = args[2];
      qmgr    = args[3];
      sslciph = args[4];
      sslkeyr = args[5];
      sslpass = args[6];
    }
    else {
      System.out.println("Usage parms: Conname Port Channel Qmgr SSLCiph SSLStore SSLKeyStorePassword");
      System.out.println("     NOTE - SSLStore is the name of a single store, which is both the keystore and truststore.");
      return;
    }
  
    new SSLSample().runSample();     
  }

  public void runSample() {
    //System.setProperty("javax.net.debug", "true");

    /****************************************************************/
    /* Utilise the arguments                                        */
    /****************************************************************/
    System.setProperty("javax.net.ssl.trustStore", sslkeyr );
    System.setProperty("javax.net.ssl.keyStore", sslkeyr );
    System.setProperty("javax.net.ssl.keyStorePassword", sslpass );
    MQEnvironment.hostname       = conname;
    MQEnvironment.port           = Integer.parseInt(port);
    MQEnvironment.channel        = channel;
    MQEnvironment.properties.put(MQConstants.SSL_CIPHER_SUITE_PROPERTY,sslciph); 

    /****************************************************************/
    /* Print out parms                                              */
    /****************************************************************/
    System.out.println("Connecting to:");
    System.out.println("  Conname = " + MQEnvironment.hostname);
    System.out.println("  Port = " + MQEnvironment.port);
    System.out.println("  Channel = " + MQEnvironment.channel);
    System.out.println("  Qmgr = " + qmgr);
    System.out.println("  SSLCiph = "+ MQEnvironment.properties.get(MQConstants.SSL_CIPHER_SUITE_PROPERTY));
    System.out.println("  SSLTrustStore = "+ System.getProperty("javax.net.ssl.trustStore"));
    System.out.println("  SSLKeyStore = "+ System.getProperty("javax.net.ssl.keyStore"));
    System.out.println("  SSLKeyStorePassword = "+ System.getProperty("javax.net.ssl.keyStorePassword"));

    try {

      /**************************************************************/
      /* Connect to queue manager                                   */
      /**************************************************************/
      System.out.println("Connecting...");
      MQQueueManager qMgr = new MQQueueManager(qmgr);
      System.out.println("Connection successful!");

      /**************************************************************/
      /* Disconnect from queue manager                              */
      /**************************************************************/
      System.out.println("Disconnecting from the Queue Manager");
      qMgr.disconnect();
      System.out.println("Done!");
    }
    catch (MQException ex) {
      System.out.println("A WebSphere MQ Error occured : Completion Code "
                + ex.completionCode + " Reason Code " + ex.reasonCode);
    }
  }
} 

File: com.ibm.mq.ssl-wizard/src/tlswizard/samples/SSLSample.c
 /*******************************************************************************
 * Copyright (c) 2005,2014 IBM Corporation and other Contributors.
 *
 * All rights reserved. This program and the accompanying materials
 * are made available under the terms of the Eclipse Public License v1.0
 * which accompanies this distribution, and is available at
 * http://www.eclipse.org/legal/epl-v10.html
 *
 * Contributors:
 *   Ian Vanstone - Initial Contribution
 *******************************************************************************/
 
 /********************************************************************/
 /*                                                                  */
 /* Program name: SSLSample                                          */
 /*                                                                  */
 /* Description: Sample C program that demonstrates how to specify   */
 /*              SSL client connection information on MQCONNX.       */
 /*                                                                  */
 /*                                                                  */
 /********************************************************************/
 /*                                                                  */
 /* Overview:                                                        */
 /*                                                                  */
 /*   This sample is provided with WebSphere MQ SupportPac MO04 -    */
 /*   WebSphere MQ SSL Wizard. The wizard will generate command line */
 /*   options to be used with this program.                          */
 /*                                                                  */
 /*   It is assumed that the SSL server connection channel and other */
 /*   SSL administration, as instructed by the wizard, has been      */
 /*   completed before running this program.                         */
 /*                                                                  */
 /*   If the SSL connection is successful the program should output: */
 /*                                                                  */
 /*      "Connection Successful!"                                    */
 /*                                                                  */
 /********************************************************************/
 /*                                                                  */
 /* Function:                                                        */
 /*                                                                  */
 /*   SSLSample is a sample C program that demonstrates how to       */
 /*   supply SSL information for a client connection on the MQCONNX  */
 /*   call.                                                          */
 /*                                                                  */
 /*   The sample simply connects to the queue manager using MQCONNX  */
 /*   and then disconnects using MQDISC.                             */
 /*                                                                  */
 /*   This program is intended to be linked as an MQI client         */
 /*   application.                                                   */
 /*                                                                  */
 /********************************************************************/
 /*                                                                  */
 /* Usage:                                                           */
 /*                                                                  */
 /*   SSLSample has 5 parameters, all of which are mandatory:        */
 /*                                                                  */
 /*     SSLSample Conname SvrconnChannelName QMgrName SSLCiph SSLKeyr*/
 /*                                                                  */
 /*   The parameters are:                                            */
 /*                                                                  */
 /*     Conname     - the connection name of the server queue        */
 /*                   manager in the same format as the CONNAME      */
 /*                   parameter on the MQSC DEFINE CHANNEL command.  */
 /*                                                                  */
 /*     SvrconnChannelName                                           */
 /*                 - the name of the server connection channel      */
 /*                   on the server queue manager with which the     */
 /*                   sample program will try to connect.            */
 /*                                                                  */
 /*     QMgrName    - the name of the server queue manager.          */
 /*                                                                  */
 /*     SSLCiph     - the SSL CipherSpec.                            */
 /*                                                                  */
 /*     SSLKeyr     - the SSL key repository.                        */
 /*                                                                  */
 /*   For example:                                                   */
 /*                                                                  */
 /*     SSLSample myhost(1414) SVRCONN1 QM1 NULL_MD5                 */
 /*                                          C:\mq\ssl\client.kdb    */
 /*                                                                  */
 /********************************************************************/
 #include <stdio.h>
 #include <stdlib.h>
 #include <ctype.h>
 #include <string.h>
 #include <cmqc.h>                  /* For regular MQI definitions   */
 #include <cmqxc.h>                 /* For MQCD definition           */

 int main(int argc, char **argv)
 {
   MQSCO    SSL_options = {MQSCO_DEFAULT};
   MQCNO    Connect_options = {MQCNO_DEFAULT};
   MQCD     ClientConn = {MQCD_CLIENT_CONN_DEFAULT};
   MQOD     od = {MQOD_DEFAULT};    
   MQCHAR   QMName[MQ_Q_MGR_NAME_LENGTH]; /* name of connection qmgr */
   MQHCONN  Hcon;                         /* connection handle       */ 
   MQLONG   CompCode;                     /* completion code         */ 
   MQLONG   Reason;                       /* reason code             */ 

   /******************************************************************/
   /* Check for correct number of arguments                          */
   /******************************************************************/
   if (argc != 6)
   {
     printf("Usage:\n");
     printf("\t%s ConnName SvrconnChannelName QMgrName SSLCiph SSLKeyr\n",
            argv[0]);
     exit(99);
   }

   /******************************************************************/
   /* Print out arguments                                            */
   /******************************************************************/
   printf("Connecting to:\n");
   printf("  Conname = %s.\n", argv[1]);
   printf("  SvrconnChannelName = %s.\n", argv[2]);
   printf("  QMgrName = %s.\n", argv[3]);
   printf("  SSLCiph = %s.\n", argv[4]);
   printf("  SSLKeyr = %s.\n", argv[5]);

   /******************************************************************/
   /* Utilise the arguments                                          */
   /******************************************************************/
   strcpy(ClientConn.ConnectionName, argv[1]);
   strcpy(ClientConn.ChannelName, argv[2]);
   strcpy(QMName, argv[3]);
   memcpy(ClientConn.SSLCipherSpec,argv[4],strlen(argv[4]));
   strcpy(SSL_options.KeyRepository, argv[5]);
   ClientConn.Version = MQCD_CURRENT_VERSION;
   Connect_options.ClientConnPtr = &ClientConn;
   Connect_options.SSLConfigPtr = &SSL_options;
   Connect_options.Version = MQCNO_VERSION_4;

   /******************************************************************/
   /* Connect to queue manager                                       */
   /******************************************************************/
   MQCONNX(QMName,                 /* queue manager                  */
           &Connect_options,       /* options for connection         */
           &Hcon,                  /* connection handle              */
           &CompCode,              /* completion code                */
           &Reason);               /* reason code                    */

   if (CompCode == MQCC_FAILED)
   {
     /* Report reason and stop if it failed                          */
     printf("MQCONNX ended with reason code %d\n", Reason);
     exit( (int)Reason );
   }
   printf("Connection Successful!.\n");

   /******************************************************************/
   /* Disconnect from queue manager                                  */
   /******************************************************************/
   MQDISC(&Hcon,                   /* connection handle              */
          &CompCode,               /* completion code                */
          &Reason);                /* reason code                    */

   if (Reason != MQRC_NONE)
   {
     /* Report reason if there is one                                */
     printf("MQDISC ended with reason code %d\n", Reason);
   }

   /******************************************************************/
   /* End of SSLSample                                               */
   /******************************************************************/
   printf("SSLSample end\n");
   return(0);
 }



